codegen/masm/src/codegen.rs

```diff
@@ -418,15 +440,9 @@ impl<'ast> AirVisitor<'ast> for CodeGenerator<'ast> {
 
     fn visit_constant_binding(
         &mut self,
-        constant: &'ast ConstantBinding,
+        _constant: &'ast ConstantBinding,
     ) -> Result<Self::Value, Self::Error> {
-        match self.constants.entry(constant.name()) {
-            Entry::Occupied(_) => Err(CodegenError::DuplicatedConstant),
-            Entry::Vacant(entry) => {
-                entry.insert(constant.value());
-                Ok(())
-            }
-        }
+        Ok(())
```

Comment:
Since this and `visit_public_input` methods don't really do anything any more, should we just get rid of them?

Reply:
I expect a correct visitor pattern implementation to walk the whole structure, that is why I left that in the trait and as a consequence left it here too.

I guess this is not being used for anything else, so it can be removed

Reply:
Yeah - I think starting with a generic approach is a fine, but as we learn more about how we need process the structure (i.e., some elements are processed at construction time, other elements may be processed together), I think it is OK to make things more specific.

Reply:
removed unused methods.